localstack/services/cloudwatch/provider_v2.py

```diff
@@ -288,60 +292,60 @@ def set_alarm_state(
             raise InvalidParameterValueException(
                 "TODO: check right error message: Json was not correctly formatted"
             )
-
-        store = self.get_store(context.account_id, context.region)
-        alarm = store.Alarms.get(
-            arns.cloudwatch_alarm_arn(
-                alarm_name, account_id=context.account_id, region_name=context.region
+        with _STORE_LOCK:
+            store = self.get_store(context.account_id, context.region)
+            alarm = store.Alarms.get(
+                arns.cloudwatch_alarm_arn(
+                    alarm_name, account_id=context.account_id, region_name=context.region
```

Comment:
q: depending on the response to the reason of the locking, if we need to keep the lock, do we need to lock the whole iteration + actually calling the actions, or could we just lock the store access? It doesn't seem the store is accessed anymore after those lines, and I guess we wouldn't need to lock the whole store for it.

Reply:
we do access the store in line 312 when calling 
```
self_update_state(...)
```
Also my intention to avoid a situation where we send wrong information about the `alarm` to sns/lambda. As we access the content of `alarm`, it could in theory be modified by another store-access before calling the actions, leading to inconsistent alarm-details. 
Maybe I am overcautions though - what are you thoughts on that?

Reply:
My take on this is that as long as we are not experiencing heavy performance degradation when locking down the store for access we should play it safe. Especially since the more fine-grained options proved more fickle than anticipated.

Reply:
Agreed, thanks for answering my questions! 😄

Reply:
as discussed with @bentsku offline: we need to see how things evolve. this approach of locking may slow things down in case we have a high number of alarm-destinations. In that case make a copy of the alarm object could be preferred. For now we leave it that way.